assembl/graphql/section.py

```diff
@@ -0,0 +1,160 @@
+
+            kwargs = {}
+            kwargs['section_type'] = args.get('section_type', 'CUSTOM')
+            kwargs['url'] = args.get('url')
+            kwargs['order'] = args.get('order')
+            saobj = cls(
+                discussion_id=discussion_id,
+                title=title_langstring,
+                **kwargs)
+            db.add(saobj)
+            db.flush()
+
+        return CreateSection(section=saobj)
+
+
+class DeleteSection(graphene.Mutation):
+    class Input:
+        section_id = graphene.ID(required=True)
+
+    success = graphene.Boolean()
+
+    @staticmethod
+    def mutate(root, args, context, info):
+        discussion_id = context.matchdict['discussion_id']
+        user_id = context.authenticated_userid or Everyone
+
+        section_id = args.get('section_id')
+        section_id = int(Node.from_global_id(section_id)[1])
+        section = models.Section.get(section_id)
+        if section.section_type != SectionTypesEnum.CUSTOM.value:
+            return DeleteSection(success=False)
+
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = section.user_can(
+            user_id, CrudPermissions.DELETE, permissions)
+        if not allowed:
+            raise HTTPUnauthorized()
+
+        section.db.delete(section)
+        section.db.flush()
+        return DeleteSection(success=True)
+
+
+class UpdateSection(graphene.Mutation):
+    class Input:
+        id = graphene.ID(required=True)
+        title_entries = graphene.List(LangStringEntryInput)
+        url = graphene.String()
+        order = graphene.Float()
+
+    section = graphene.Field(lambda: Section)
+
+    @staticmethod
+    def mutate(root, args, context, info):
+        cls = models.Section
+        discussion_id = context.matchdict['discussion_id']
+        user_id = context.authenticated_userid or Everyone
+        section_id = args.get('id')
+        section_id = int(Node.from_global_id(section_id)[1])
+        section = cls.get(section_id)
+
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = section.user_can(
+            user_id, CrudPermissions.UPDATE, permissions)
+        if not allowed:
+            raise HTTPUnauthorized()
+
+        with cls.default_db.no_autoflush as db:
+            title_entries = args.get('title_entries')
+            if title_entries is not None and len(title_entries) == 0:
+                raise Exception(
+                    'section titleEntries needs at least one entry')
+                # Better to have this message than
+                # 'NoneType' object has no attribute 'owner_object'
+                # when creating the saobj below if title=None
+
+            update_langstring_from_input_entries(
+                section, 'title', title_entries)
+
+            kwargs = {}
+            kwargs['url'] = args.get('url', None)
+            kwargs['order'] = args.get('order', None)
```

Comment:
same here, order shouldn't be None.

Reply:
as it is an update, I thought that I should enable None and that it will not modify it

Reply:
done

Reply:
FYI `args.get('order', None)` and `args.get('order')` is actually the same, it's the dict api. If you want to not specify order and keep the current value, you need to add a condition on `args.get('order')` before setting `kwargs['order'] = None`. Now it will raise an error because you added nullable=False.